pkg/providers/subnet/subnet.go

```diff
@@ -201,11 +204,22 @@ func (p *Provider) UpdateInflightIPs(createFleetInput *ec2.CreateFleetInput, cre
 
 func (p *Provider) LivenessProbe(req *http.Request) error {
 	p.Lock()
-	//nolint: staticcheck
+	if err := p.checkAWSAuth(req.Context()); err != nil {
+		return err
+	}
 	p.Unlock()
 	return nil
 }
 
+func (p *Provider) checkAWSAuth(ctx context.Context) error {
+	_, err := p.ec2api.DescribeSubnetsWithContext(ctx, &ec2.DescribeSubnetsInput{DryRun: aws.Bool(true)})
+	var aerr awserr.Error
+	if errors.As(err, &aerr) && aerr.Code() == errcode.DryRunOperation {
```

Comment:
Maybe consider creating a `IsDryRunOperationError` and `IgnoreDryRunOperationError` similar to the funcs provided in the client-go errors package.